### PR TITLE
Adds URL Field Type

### PR DESCRIPTION
**Pull request by freshleafmedia**
## Description

Simply adds `url` as a field type.

As noted in a PR #1513 I am unable to build the dist assets due to errors in unrelated files

### CODE REVIEW

File: frontend/js/components/Textfield.vue [hunk 1]
```diff
@@ -87,6 +87,23 @@
         @blur="onBlur"
         @input="onInput"
       />
+      <input v-if="type === 'url'"
```
Comment (haringsrob) [inline frontend/js/components/Textfield.vue:90]: Thinking as most of these are the same we could just do:

```vue
<input
        ref="input"
        :type="type"
        :placeholder="placeholder"
        :name="name"
        :id="uniqId"
        :disabled="disabled"
        :maxlength="displayedMaxlength"
        :required="required"
        :readonly="readonly"
        :autofocus="autofocus"
        :autocomplete="autocomplete"
        :value="value"
        @focus="onFocus"
        @blur="onBlur"
        @input="onInput"
      />
```

If it is not the e-mail (or even then we can conditionally add the pattern.).

**Comment by haringsrob** [comment]
Hey @freshleafmedia!

Thanks for the contribution, I had a quick test and it builds fine here. Maybe it is a difference in node version?

I also left a suggestion, what do you think about that?

File: frontend/js/components/Textfield.vue (hunk 1 above)
Comment (freshleafmedia) [reply frontend/js/components/Textfield.vue:90]: I had the same thought, I can create a separate PR for that :+1:

**Comment by freshleafmedia** [comment]
Hi @haringsrob

Oh possibly, I'm using Node v16.14.0 what are you using?

**Comment by haringsrob** [comment]
I am on v17.6.0

**Pull request merged**